blog/2023-03-28-how-to-use-witnet-oracle-to-display-cryptocurrency-prices/2023-03-28-how-to-use-witnet-oracle-to-display-cryptocurrency-prices.md

```diff
@@ -0,0 +1,277 @@
+
+The `value` key represents a unique identifier for simplifying the identification of currency pairs served by Witnet's Price Router. One could use only the first 4 bytes of the full bytes32 identifier (ID32), which are referred to as ID4.
+
+### helpers.js
+
+```js
+export const truncate = input => `${input.substring(0, 5)}...${input.slice(-4)}`
+```
+
+`truncate` receives a string as input, shortens it to a length of 9 characters by taking the first five characters, adding an ellipsis ("..."), and then appending the last four characters of the string. Finally, it returns the shortened string.
+
+### interact.js
+
+```js
+
+import { ID4s } from './constants'
+
+export const priceFeedAddress = '0xB40bf553be448345C09891682636cf632a77c419'
+
+export const getPriceFeed = async (contract) => {
+
+  if (contract) {
+    try {
+      let values = ID4s.map(item => item.value)
+      values = values.map(item => '0x' + item.toString(16));
+
+      let priceFeeds = []
+
+      for (let i = 0; i < ID4s.length; i++) {
+        const priceFeed = new Promise(async (resolve) => {
+
+          const priceFeed = await contract.methods.getPriceFeedLastValues(values[i]).call()
+          resolve({
+            price: (priceFeed._lastPrices / 1000000).toFixed(2),
+            caption: ID4s[i].caption
+          })
+        })
+        priceFeeds.push(priceFeed)
+      }
+      return Promise.all(priceFeeds)
+    } catch (e) {
+      console.log({e})
+    }
+  }
+}
+```
+The `getPriceFeed` function takes a single argument, `contract`. It's an asynchronous function that performs a series of steps when the contract argument is truthy. First, it creates an array of the value properties from the ID4s constant, converts each value to a hexadecimal string, and pushes each string to the `values` array.
+
+Next, it creates an empty array called `priceFeeds` and loops through the ID4s array. For each item in ID4s, it creates a new Promise that awaits the result of calling the `getPriceFeedLastValues` method on the contract object with the corresponding value as an argument.
+
+When the `getPriceFeedLastValues` method call completes, the Promise resolves with an object containing the `price` value and the `caption` from the current ID4s item. The resolved Promise is then pushed to the priceFeeds array.
+
+## Conclusion
+
+We built an NFT marketplace for tech artisans to connect with their prospective clients with a metric to highlight freelancers with frequent sales. [Here](https://witnet-oracle.vercel.app/) is the final demo of what we built. [Here](https://github.com/cjustinobi/witnet-oracle) is the link to the complete code.
+
+## About the Author​
+
+A software engineer, co-founder, Africinnovate, and a Web3 enthusiast. I used to call myself VueDetective. Connect with me on LinkedIn and Github
+
+## References​
+
+[https://docs.witnet.io/smart-contracts/witnet-data-feeds](https://docs.witnet.io/smart-contracts/witnet-data-feeds)
```

Comment:
Can you correct this link give it a better description than using the link itself as the description